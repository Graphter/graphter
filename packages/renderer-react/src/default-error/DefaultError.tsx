import { ErrorRendererProps } from "../error-boundary/ErrorBoundary";
import React  from "react";

export default function DefaultError({ err }: ErrorRendererProps){
  if(typeof err === 'string') return <div data-testid='error'>error: {err}</div>;
  else if('message' in err) return <div data-testid='error'>error: {err.message}</div>;
  return <div data-testid='error'>error: unreadable error</div>;
}