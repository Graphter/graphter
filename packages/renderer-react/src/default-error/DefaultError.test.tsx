import React from "react";
import { act, render, within, fireEvent } from '@testing-library/react';
import { screen } from '@testing-library/dom'
import '@testing-library/jest-dom/extend-expect'
import DefaultError from "./DefaultError";

describe(`<DefaultError />`, () => {
  it(`should render string errors`, () => {
    render(<DefaultError err={'This is a string error'} />);
    expect(screen.getByTestId('error').textContent).toMatchSnapshot();
  });
  it(`should render Error messages`, () => {
    render(<DefaultError err={new Error('This is an Error error')} />);
    expect(screen.getByTestId('error').textContent).toMatchSnapshot();
  });
  it(`should fallback if incorrect error param is passed`, () => {
    // @ts-ignore
    render(<DefaultError err={{ foo: 'bar' }} />);
    expect(screen.getByTestId('error').textContent).toMatchSnapshot();
  });
});

export {};