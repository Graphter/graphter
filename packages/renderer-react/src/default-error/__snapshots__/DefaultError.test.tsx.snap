// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultError /> should fallback if incorrect error param is passed 1`] = `"error: unreadable error"`;

exports[`<DefaultError /> should render Error messages 1`] = `"error: This is an Error error"`;

exports[`<DefaultError /> should render string errors 1`] = `"error: This is a string error"`;
